x-pack/plugins/ml/public/application/util/timeseriesexplorer_utils.ts

```diff
@@ -0,0 +1,649 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { IUiSettingsClient } from '@kbn/core/public';
+import { aggregationTypeTransform } from '@kbn/ml-anomaly-utils';
+import { isMultiBucketAnomaly, ML_JOB_AGGREGATION } from '@kbn/ml-anomaly-utils';
+import { extractErrorMessage } from '@kbn/ml-error-utils';
+import moment from 'moment';
+import { forkJoin, Observable, of } from 'rxjs';
+import { each, get } from 'lodash';
+import { catchError, map } from 'rxjs/operators';
+import { type MlAnomalyRecordDoc } from '@kbn/ml-anomaly-utils';
+import { parseInterval } from '../../../common/util/parse_interval';
+import type { GetAnnotationsResponse } from '../../../common/types/annotations';
+import { mlFunctionToESAggregation } from '../../../common/util/job_utils';
+import { ANNOTATIONS_TABLE_DEFAULT_QUERY_SIZE } from '../../../common/constants/search';
+import { CHARTS_POINT_TARGET } from '../timeseriesexplorer/timeseriesexplorer_constants';
+import { timeBucketsServiceFactory } from './time_buckets_service';
+import type { TimeRangeBounds } from './time_buckets';
+import type { Job } from '../../../common/types/anomaly_detection_jobs';
+import type { TimeBucketsInterval } from './time_buckets';
+import type {
+  ChartDataPoint,
+  FocusData,
+  Interval,
+} from '../timeseriesexplorer/timeseriesexplorer_utils/get_focus_data';
+import type { CriteriaField } from '../services/results_service';
+import {
+  MAX_SCHEDULED_EVENTS,
+  TIME_FIELD_NAME,
+} from '../timeseriesexplorer/timeseriesexplorer_constants';
+import type { MlApiServices } from '../services/ml_api_service';
+import { mlResultsServiceProvider, type MlResultsService } from '../services/results_service';
+import { forecastServiceProvider } from '../services/forecast_service_provider';
+import { timeSeriesSearchServiceProvider } from '../timeseriesexplorer/timeseriesexplorer_utils/timeseries_search_service_provider';
+import { useMlKibana } from '../contexts/kibana';
+
+export function timeSeriesExplorerServiceFactory(
```

Comment:
Suggest to rename the file to `time_series_explorer_service.ts` to be in line with how we renamed the time buckets services.